docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,157 @@
+
+### 2.3 Managed Code
+Since ML.NET has a hard dependency on Intel MKL, the managed code imports the dlls without checking whether or not they exist. If the dlls don't exist you get a hard failure. For example, the base test class imports and sets up Intel MKL even if the test itself does not need it. This same situation applies to our own native dlls. When they aren't present, the imports fail. We will need to make ML.NET correctly handle the dll imports.
+
+### 2.4 Native Projects
+ML.NET has 6 native projects. They are:
+ - CpuMathNative
+   - CpuMathNative relies heavily on IntelMKL. Using NetCore 3.1, however, there is a full software fallback.
+ - FastTreeNative
+   - FastTreeNative has a software fallback using a C# compiler flag. This flag is hardcoded to always use the native code.
+ - LdaNative
+   - Builds successfully without native dependencies.
+ - MatrixFactorizationNative
+   - Uses libmf. Currently, we are hardcoding the "USESSE" flag which requires Intel MKL. Removing this flag allows MatrixFactorizationNative to build for Arm.
+ - MklProxyNative
+   - Wrapper for Intel MKL. When Intel MKL is not present, this is not needed. However, the build is hardcoded to always compile this code.
+ - SymSgdNative
+   - Only uses 4 Intel MKL methods. For 2 of them, I can't find direct replacements on other architectures.
+
+Of these 6, only LdaNative builds successfully for other architectures without changing anything.
+
+### 2.5 3rd Party Dependencies
+As mentioned above, there are several 3rd party packages that don't have support for non x86/x64 machines.
+ - LightGBM. LightGBM doesn't offer packages for non x86/x64. I was able to build the code for Arm64, but we would either have to build it ourselves, convince them to create more packages for us, or annotate that this doesn't work on non x86/64 machines.
+ - TensorFlow. The full version of TensorFlow only runs on x86/x64. There is a [lite](https://www.tensorflow.org/lite/guide/build_arm64) version that supports Arm64, and you can install it directly with python, but this isn't the full version so not all models will run. We would also have to verify if the C# library we use to interface with TensorFlow will work with the lite version.
+ - OnnxRuntime. OnnxRuntime doesn't have prebuilt packages for more than just x86/x64. It does support Arm, but we have to build it ourselves. This is the same situation as with LightGBM.
+
+
+## 3 Possible Solutions
+We have several possible options we can use to resolve this:
+ - Fully remove Intel MKL as a dependency and use software that supports all platforms/architectures.
```

Comment:
Rather than focus on Intel MKL here, can we rephrase to just say something like
 - Keep ML.NET native components and rewrite to a avoid problematic dependencies completely
 - Keep ML.NET native components and ifdef/rewrite to avoid problematic dependencies on platforms/architectures only where they are not supported.
 - Eliminate ML.NET native components and implement all functionality in managed code.
 - Hybrid